webapp/src/widgets/editableDayPicker.tsx

```diff
@@ -1,26 +1,83 @@
 import {useIntl} from 'react-intl'
-import DayPickerInput from 'react-day-picker/DayPickerInput'
+import DayPicker, {DateUtils} from 'react-day-picker'
 import MomentLocaleUtils from 'react-day-picker/moment'
 
+import Editable from '../widgets/editable'
+import SwitchOption from '../widgets/menu/switchOption'
+import Button from '../widgets/buttons/button'
+
+import Modal from '../components/modal'
+import ModalWrapper from '../components/modalWrapper'
+
 import 'react-day-picker/lib/style.css'
 import './editableDayPicker.scss'
+import {Utils} from '../utils'
 
 type Props = {
     className: string
     value: string
-    dateFormat: string
-    onChange: (value: string | undefined) => void
+    onChange: (value: string) => void
+}
+
+type DateProperty = {
+    from?: number
+    to?: number
+    includeTime?: boolean
+    timeZone?: string
 }
 
 const loadedLocales: Record<string, any> = {}
 
 function EditableDayPicker(props: Props): JSX.Element {
-    const {className, onChange, dateFormat} = props
+    const {className, value, onChange} = props
     const intl = useIntl()
-    const [value, setValue] = useState(props.value ? new Date(parseInt(props.value, 10)) : undefined)
+    const timeZoneOffset = new Date().getTimezoneOffset() * 60 * 1000
+
+    const getDisplayDate = (date: Date | null | undefined) => {
+        let displayDate = ''
+        if (date) {
+            displayDate = Utils.displayDate(date, intl)
+        }
+        return displayDate
+    }
+
+    const createDatePropertyFromString = (initialValue: string) => {
+        let dateProperty: DateProperty = {}
+        if (initialValue) {
+            const singleDate = new Date(Number(initialValue))
+            if (singleDate && DateUtils.isDate(singleDate)) {
+                dateProperty.from = singleDate.getTime()
+            } else {
+                dateProperty = JSON.parse(initialValue)
+                if (!dateProperty.includeTime) {
+                    // if date only, convert from UTC to local time.
+                    if (dateProperty.from) {
+                        dateProperty.from += timeZoneOffset
+                    }
+                    if (dateProperty.to) {
+                        dateProperty.to += timeZoneOffset
+                    }
+                }
+            }
+        }
+        return dateProperty
+    }
+
+    const [dateProperty, setDateProperty] = useState<DateProperty>(createDatePropertyFromString(value as string))
+    const [showDialog, setShowDialog] = useState(false)
+
+    const dateFrom = dateProperty.from ? new Date(dateProperty.from) : undefined
+    const dateTo = dateProperty.to ? new Date(dateProperty.to) : undefined
+    const [fromInput, setFromInput] = useState<string>(getDisplayDate(dateFrom))
+    const [toInput, setToInput] = useState<string>(getDisplayDate(dateTo))
+
+    const stateRef = useRef(dateProperty)
+    stateRef.current = dateProperty
+
+    const isRange = dateTo !== undefined
```

Comment:
This trick work here because you are not changing the properties on the fly, but if in the future it can happen, you will end up with a wrong `isRange` inside the callbacks, so this line is perfectly fine, but you would need to generate the callbacks using `useCallback` and adding the right dependencies, en this case can be the `isRange`, or maybe the `datePropertype`.